core/src/main/java/io/grpc/NameResolverProvider.java

```diff
@@ -49,6 +49,8 @@
  */
 @Internal
 public abstract class NameResolverProvider extends NameResolver.Factory {
+  /** Only visible to be used in ManagedChannelImpl. Do not use. */
+  public static final String NO_PROVIDERS_SCHEME = "noproviders";
```

Comment:
Since it's not meant to be used as a scheme, can we use some special characters to make it invalid, e.g., `*noproviders*`?